### PR TITLE
Allow anonymous user colors

### PR DESCRIPTION
**Pull request by davidmehren**
### Component/Part
database schema + entities

### Description
This PR changes the database schema to allow to save the edits of anonymous users.
This is being done by introducing a new `Author` entity, that either represents a `User` (when the edit was made logged-in) or a `Session` (when the edit was made without a login).

Notes created by anonymous users don't have an `owner`, respective changes were made.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: src/revisions/authorship.entity.ts [hunk 1]
```diff
@@ -13,11 +13,11 @@ import {
...
-import { User } from '../users/user.entity';
+import { Author } from '../authors/author.entity';
 import { Revision } from './revision.entity';
 
 /**
- * This class stores which parts of a revision were edited by a particular user.
+ * The Authorship represents a change in the content of a note by a particular {@link Author}
  */
 @Entity()
 export class Authorship {
```
Comment (DerMolly) [inline src/revisions/authorship.entity.ts:23]: Didn't you plan to rename this to Edit?

**Review by InnayTool — APPROVED** [review]

File: src/notes/notes.service.spec.ts [hunk 2]
```diff
@@ -45,6 +46,7 @@ describe('NotesService', () => {
   let forbiddenNoteId: string;
 
   beforeEach(async () => {
+    userRepo = new Repository<User>();
```
Comment (DerMolly) [inline src/notes/notes.service.spec.ts:49]: Please add an explanation why you do this here?

**Review by DerMolly — APPROVED** [review]

File: src/revisions/authorship.entity.ts (hunk 1 above)
Comment (davidmehren) [reply src/revisions/authorship.entity.ts:23]: Not in this PR

File: src/notes/notes.service.spec.ts (hunk 2 above)
Comment (davidmehren) [reply src/notes/notes.service.spec.ts:54]: done

**Review by DerMolly — APPROVED** [review]

**Comment by InnayTool** [comment]
closes #505.

**Pull request merged**